backend/de.metas.postfinance/src/main/java/de/metas/postfinance/document/export/impl/DunningPostFinanceYbInvoiceHandler.java

```diff
@@ -0,0 +1,139 @@
+import de.metas.document.DocBaseAndSubType;
+import de.metas.document.DocTypeId;
+import de.metas.document.IDocTypeDAO;
+import de.metas.document.engine.DocStatus;
+import de.metas.dunning.DunningDocId;
+import de.metas.dunning.api.IDunningDAO;
+import de.metas.dunning.export.DunningToExportFactory;
+import de.metas.dunning.model.I_C_DunningDoc;
+import de.metas.dunning_gateway.spi.model.DunningToExport;
+import de.metas.invoice.InvoiceId;
+import de.metas.invoice.export.InvoiceToExportFactory;
+import de.metas.invoice.service.IInvoiceBL;
+import de.metas.invoice_gateway.spi.model.export.InvoiceToExport;
+import de.metas.postfinance.document.export.IPostFinanceYbInvoiceHandler;
+import de.metas.postfinance.document.export.PostFinanceDocumentType;
+import de.metas.postfinance.document.export.PostFinanceExportException;
+import de.metas.postfinance.document.export.PostFinanceYbInvoiceRequest;
+import de.metas.postfinance.document.export.PostFinanceYbInvoiceResponse;
+import de.metas.postfinance.document.export.PostFinanceYbInvoiceService;
+import de.metas.postfinance.jaxb.Invoice;
+import de.metas.postfinance.ybinvoice.v2.Envelope;
+import de.metas.postfinance.ybinvoice.v2.FixedReference;
+import de.metas.util.Services;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.adempiere.util.lang.impl.TableRecordReference;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Nullable;
+import javax.xml.datatype.XMLGregorianCalendar;
+import java.util.GregorianCalendar;
+import java.util.List;
+import java.util.Optional;
+
+import static de.metas.postfinance.document.export.PostFinanceYbInvoiceService.YB_INVOICE_OBJECT_FACTORY;
+
+@Component
+@RequiredArgsConstructor
+public class DunningPostFinanceYbInvoiceHandler implements IPostFinanceYbInvoiceHandler
+{
+	private final IDocTypeDAO docTypeDAO = Services.get(IDocTypeDAO.class);
+	private final IDunningDAO dunningDAO = Services.get(IDunningDAO.class);
+	private final IInvoiceBL invoiceBL = Services.get(IInvoiceBL.class);
+
+	@NonNull private final DunningToExportFactory dunningToExportFactory;
+	@NonNull private final InvoiceToExportFactory invoiceToExportFactory;
+	@NonNull private final PostFinanceYbInvoiceService postFinanceYbInvoiceService;
+	@Override
+	public boolean applies(final PostFinanceYbInvoiceRequest postFinanceYbInvoiceRequest)
+	{
+		final TableRecordReference documentReference = postFinanceYbInvoiceRequest.getDocumentReference();
+		if(!documentReference.getTableName().equals(I_C_DunningDoc.Table_Name))
+		{
+			return false;
+		}
+
+		final I_C_DunningDoc dunningRecord = dunningDAO.getByIdInTrx(DunningDocId.ofRepoId(documentReference.getRecord_ID()));
+		final DocBaseAndSubType docBaseAndSubType = docTypeDAO.getDocBaseAndSubTypeById(DocTypeId.ofRepoId(dunningRecord.getC_DocType_ID()));
+		return DocStatus.ofCode(dunningRecord.getDocStatus()).isCompleted() && docBaseAndSubType.getDocBaseType().isDunningDoc();
+	}
+
+	@Override
+	@Nullable
```

Comment:
why `@Nullable` ?

Reply:
leftover from refactor, used return null instead of exception at the beginning